### ISSUE TITLE
Action is not authorized with current scopes; requires any of [delete:servers] when deleting VL

### ISSUE THREAD

**Issue by gpelouze** [QCDIS/k8s-jhub#6]
![Image](https://user-images.githubusercontent.com/9514702/267043739-7c26f2ae-2e2e-46f1-92b9-6d4030d2e414.png)

**Comment by gpelouze** [comment]
Related https://discourse.jupyter.org/t/action-is-not-authorized-with-current-scopes-requires-any-of-read-servers/16685/6

**Comment by gpelouze** [comment]
VL can be deleted by logging out

**Comment by gpelouze** [comment]
After updating to Jupyter Hub 4.0.2 (https://github.com/QCDIS/infrastructure/issues/65), the issue becomes:

![Image](https://github.com/QCDIS/NaaVRE/assets/9514702/fb748272-8cbb-42aa-8fb5-6473ac26c08f)

(Image was deleted during migration. Error message was "'_xsrf' argument missing from DELETE")

**Comment by skoulouzis** [comment]
As suggested by: https://saturncloud.io/blog/how-to-fix-a-jupyter-notebook-not-saving-xsrf-argument-missing-from-post/ 
We may need to reset the Jupyter API access token. However, when we start the Jupyter lab we get this error in the  user's pod (jupyter-USER)

```
[E 2023-09-15 14:10:35.470 ServerApp] Uncaught exception GET /vl-vol2bird/user/USER/github?1694787035365 (10.244.0.0)
    HTTPServerRequest(protocol='http', host='', method='GET', uri='/vl-vol2bird/user/USER/github?1694787035365', version='HTTP/1.1', remote_ip='10.244.0.0')
    Traceback (most recent call last):
      File "/venv/lib/python3.11/site-packages/tornado/web.py", line 1786, in _execute
        result = await result
                 ^^^^^^^^^^^^
      File "/venv/lib/python3.11/site-packages/tornado/gen.py", line 234, in wrapper
        yielded = ctx_run(next, result)
                  ^^^^^^^^^^^^^^^^^^^^^
      File "/venv/lib/python3.11/site-packages/jupyterlab_github/__init__.py", line 95, in get
        headers={"Authorization": "token {}".format(token)}
                                                    ^^^^^
    UnboundLocalError: cannot access local variable 'token' where it is not associated with a value
[W 2023-09-15 14:10:35.472 ServerApp] Unhandled error
```

**Comment by skoulouzis** [comment]
@gpelouze isn't this an issue for the PaaS ?

**Comment by gpelouze** [comment]
The xsrf issue comes from our custom page templates ([QCDIS/k8s-jhub](https://github.com/QCDIS/k8s-jhub)), as explained here: https://discourse.jupyter.org/t/jupyterhub-deployment-on-k8s-1-27/19646/16

When disabling the custom pages (commenting-out `initContainers`, `extraVolumes`, `extraVolumeMounts`, and `extraConfig` in the helm chart's values), the issue disappears. 

The issue can probably be fixed by updating https://github.com/QCDIS/k8s-jhub/tree/master/templates using https://github.com/jupyterhub/jupyterhub/tree/3d60ad3956bf5f39dc3b3f762d530374668ce8ef/share/jupyterhub/templates

**Comment by gpelouze** [comment]
Closed by #7
